tests/models/test_pyro.py

```diff
@@ -0,0 +1,196 @@
+        #     .to_event(2)
+        # )
+        # self.linear.bias = PyroSample(
+        #     dist.Normal(self.zero, self.ten).expand([out_features]).to_event(1)
+        # )
+
+    @staticmethod
+    def _get_fn_args_from_batch(tensor_dict):
+        x = tensor_dict[_CONSTANTS.X_KEY]
+        y = tensor_dict[_CONSTANTS.LABELS_KEY]
+
+        return (x, y.squeeze(1)), {}
+
+    def forward(self, x, y):
+        sigma = pyro.sample("sigma", dist.Uniform(self.zero, self.ten))
+        mean = self.linear(x).squeeze(-1)
+        with pyro.plate("data", x.shape[0]):
+            pyro.sample("obs", dist.Normal(mean, sigma), obs=y)
+        return mean
+
+    def guide(self, x, y):
+        return self._auto_guide(x, y)
+
+
+class SCVIPyro(PyroModule, PyroBaseModuleClass):
+    def __init__(self, n_input, n_latent):
+
+        super().__init__()
+        self.n_input = n_input
+        self.n_latent = n_latent
+        self.epsilon = 5.0e-3
+        # z encoder goes from the n_input-dimensional data to an n_latent-d
+        # latent space representation
+        self.encoder = Encoder(
+            n_input,
+            n_latent,
+            n_layers=1,
+            n_hidden=128,
+            dropout_rate=0.1,
+        )
+        # decoder goes from n_latent-dimensional space to n_input-d data
+        self.decoder = DecoderSCVI(
+            n_latent,
+            n_input,
+            n_layers=1,
+            n_hidden=128,
+        )
+        # This gene-level parameter modulates the variance of the observation distribution
+        self.px_r = torch.nn.Parameter(torch.ones(self.n_input))
+
+    @staticmethod
+    def _get_fn_args_from_batch(tensor_dict):
+        x = tensor_dict[_CONSTANTS.X_KEY]
+        log_library = torch.log(torch.sum(x, dim=1, keepdim=True) + 1e-6)
+        return (x, log_library), {}
+
+    def forward(self, x, log_library):
+        # register PyTorch module `decoder` with Pyro
+        # pyro.module("scvi", self)
+        with pyro.plate("data", x.shape[0]):
+            # setup hyperparameters for prior p(z)
+            z_loc = x.new_zeros(torch.Size((x.shape[0], self.n_latent)))
+            z_scale = x.new_ones(torch.Size((x.shape[0], self.n_latent)))
+            # sample from prior (value will be sampled by guide when computing the ELBO)
+            z = pyro.sample("latent", dist.Normal(z_loc, z_scale).to_event(1))
+            # decode the latent code z
+            px_scale, _, px_rate, px_dropout = self.decoder("gene", z, log_library)
+            # build count distribution
+            nb_logits = (px_rate + self.epsilon).log() - (
+                self.px_r.exp() + self.epsilon
+            ).log()
+            x_dist = dist.ZeroInflatedNegativeBinomial(
+                gate_logits=px_dropout, total_count=self.px_r.exp(), logits=nb_logits
+            )
+            # score against actual counts
+            pyro.sample("obs", x_dist.to_event(1), obs=x)
+
+    @pyro_method
```

Comment:
See note above: a `PyroModule` model cannot own its guide; you'll need an outer `nn.Module` to contain both.